packages/Testing/PHPUnit/AbstractTestCase.php

```diff
@@ -59,8 +60,17 @@ protected function getService(string $type): object
             );
         }
 
-        $object = self::$currentContainer->get($type);
+        try {
+            $object = self::$currentContainer->get($type);
+        } catch (Throwable $e) {
+            // clear compiled container cache, to trigger re-discovery
+            RectorKernel::clearCache();
+
+            throw $e;
+        }
```

Comment:
This seems hacky. What is it need for and when it happens? It might be hiding a bug we have

We can adress it later, to keep this PR merged :+1: